lib/scanner.js

```diff
@@ -26,14 +26,30 @@ module.exports = {
   hasCamera
 }
 
-function setConfig (width, height, cam) {
-  const format = cam.formats.filter(f => f.formatName === 'MJPG' &&
-    f.width === width &&
-    f.height === height
-  )[0]
+const maxCamResolutions = []
+const minCamResolutions = []
+
+function setConfig (cam) {
+  var format = null
+
+  const pixelRes = format => format.width * format.height
+  const sortByPixelRes = _.sortBy(pixelRes)
+  const isSuitableRes = currentRes => {
+    const isUnderMaxAcceptableResolutions = _.map(res => pixelRes(currentRes) < pixelRes(res), maxCamResolutions).every(bool => bool === true)
+    const isAboveMinAcceptableResolutions = _.map(res => pixelRes(currentRes) > pixelRes(res), minCamResolutions).every(bool => bool === true)
+    return isUnderMaxAcceptableResolutions && isAboveMinAcceptableResolutions
+  }
+
+  const availableFormats = sortByPixelRes(cam.formats.filter(f => f.formatName === 'MJPG'))
+
+  _.map(elem => {
+    if (isSuitableRes(elem)) format = elem
+  }, availableFormats)
```

Comment:
How is this picking the best resolution? Isn't it going to be random?

We should get the best possible resolution and cache it, we'll be using that every time so no need to run this again.

Reply:
About the questions. We sort the available formats in ascending order, we will get the best suitable res.
But this can be optimized.